spec/controllers/supplejack_api/harvester/partners_controller_spec.rb

```diff
@@ -0,0 +1,121 @@
+# The majority of the Supplejack API code is Crown copyright (C) 2014, New Zealand Government, 
+# and is licensed under the GNU General Public License, version 3.
+# One component is a third party component. See https://github.com/DigitalNZ/supplejack_api for details. 
+# 
+# Supplejack was created by DigitalNZ at the National Library of NZ and 
+# the Department of Internal Affairs. http://digitalnz.org/supplejack
+
+require 'spec_helper'
+
+module SupplejackApi
+  describe Harvester::PartnersController, type: :controller do
+    routes { SupplejackApi::Engine.routes }
+
+    context 'with a api_key with harvester role' do
+      let(:api_key) { create(:user, role: 'harvester').api_key }
+
+      before do
+        allow(RecordSchema).to receive(:roles) { { harvester: double(:harvester, harvester: true) } }
+      end
+
+      describe "POST 'create'" do
+        it "creates a new partner" do
+          expect(Partner).to receive(:create).with("name" => "Statistics New Zealand").and_call_original
+          post :create, partner: FactoryGirl.attributes_for(:partner), api_key: api_key
+          expect(response).to be_success
```

Comment:
Can we check that the count of Partners goes up?

Reply:
done thanks !